pkg/apis/internal.acorn.io/v1/event.go

```diff
@@ -21,6 +21,15 @@ type EventInstanceList struct {
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 type EventInstance struct {
+	EventCommon `json:",inline"`
+
+	// App is the name of the app that the event is related to, if any.
+	// +optional
+	App string `json:"app,omitempty"`
```

Comment:
This approach feels very kludgy. Is there an alternative that I missed?

My instinct is that `EventSource` -- to be renamed `EventResource` soon -- can indicate the app the event is related to, if any. If we make it into a list it can track other related objects too; e.g. App and ContainerReplica.

but maybe I'm missing something obvious

Reply:
The field should actually be called `AppName`.  I think the field is useful, overly generic data structures are generically useless.

Reply:
Got it. Thanks for the guidance there.